oxide-auth-db/src/db_service/redis.rs

```diff
@@ -0,0 +1,152 @@
+use r2d2_redis::RedisConnectionManager;
+use r2d2::Pool;
+use std::env;
+use r2d2_redis::redis::Commands;
+use oxide_auth::primitives::registrar::{EncodedClient, ClientType};
+use crate::primitives::db_registrar::OauthClientDBRepository;
+use dotenv::dotenv;
+use url::Url;
+use oxide_auth::primitives::prelude::Scope;
+use std::str::FromStr;
+
+pub const REDIS_POOL_SIZE: u32 = 32;
+
+/// redis datasource to Client entries.
+
+#[derive(Debug, Clone)]
+pub struct RedisDataSource {
+    pub url: String,
+    pub pool: Pool<RedisConnectionManager>,
+}
+
+impl RedisDataSource {
+
+    pub fn new() -> Self {
+        dotenv().ok();
+        let url = env::var("REDIS_URL").expect("REDIS_URL must be set");
+        let pool = REDIS_POOL.clone();
+        RedisDataSource { url, pool }
+    }
+
+    pub fn get_url(&self) -> String {
+        self.url.to_string()
+    }
+    pub fn get_pool(self) -> Pool<RedisConnectionManager> {
+        self.pool
+    }
+}
+
+
+lazy_static! {
+    pub static ref REDIS_POOL: r2d2::Pool<r2d2_redis::RedisConnectionManager> = {
+        dotenv::dotenv().ok();
+        let redis_url = std::env::var("REDIS_URL").expect("REDIS_URL must be set");
+        let manager = r2d2_redis::RedisConnectionManager::new(redis_url).unwrap();
+        let max_pool_size: u32 = env::var("REDIS_POOL_SIZE")
+            .unwrap_or_else(|_| REDIS_POOL_SIZE.to_string())
+            .parse::<u32>()
+            .unwrap_or(REDIS_POOL_SIZE);
+
+        r2d2::Pool::builder()
+            .max_size(max_pool_size)
+            .build(manager)
+            .expect("Failed to create redis pool.")
+    };
+
+}
```

Comment:
Here as well. I see the motivation but there are alternatives. Firstly, you could provide a wrapper that makes it easy to define this static, i.e. encapsulate this as a single method method. If that doesn't work then it could also be a feature that can be en-/disabled.

Reply:
Yeah, I have also refactored these.